test/cppsuite/src/main/thread_worker.cpp

```diff
@@ -85,6 +85,8 @@ thread_worker::thread_worker(uint64_t id, thread_type type, configuration *confi
         op_track_cursor = session.open_scoped_cursor(op_tracker->get_operation_table_name());
 
     testutil_assert(key_size > 0 && value_size > 0);
+    if (type == thread_type::INSERT)
```

Comment:
Can we have a comment here please?
And what if I have more threads for the populate phase?

Reply:
Comment added.
The logic for populate phase and insert_operation phase are the same, so I think we should include `testutil_assert( db.get_collection_count() >= thread_count)` in populate phase as well, since more threads won't help speed up the test. What's your opinion?

Reply:
Yes I agree.

Reply:
Sorry but the code was not correct -- only those tests having the logic of distributing collections evenly to threads will need the assert, and there are other tests which do not have this logic. I will remove the assert in thread_worker constructor and add it to the target tests.